extensions/src/main/java/org/pentaho/platform/plugin/services/email/EmailService.java

```diff
@@ -186,18 +242,39 @@ protected PasswordAuthentication getPasswordAuthentication() {
       msg.setText( messages.getString( "EmailService.MESSAGE" ) );
       msg.setHeader( "X-Mailer", "smtpsend" );
       msg.setSentDate( new Date() );
-      Transport.send( msg );
-      sendEmailMessage = "EmailTester.SUCESS";
+      sendEmail( emailConfig, token, session, msg );
+      sendEmailMessage = TEST_EMAIL_SUCCESS;
     } catch ( Exception e ) {
       logger.error( messages.getString( "EmailService.NOT_CONFIGURED" ), e );
-      sendEmailMessage = "EmailTester.FAIL";
+      sendEmailMessage = TEST_EMAIL_FAIL;
     }
+    setEmailConfig( emailConfig );
     return sendEmailMessage;
   }
 
+  public void sendEmail( final IEmailConfiguration emailConfig, IEmailAuthenticationResponse token, Session session, MimeMessage msg ) throws MessagingException, IOException, HttpException {
+    if ( emailConfig.getAuthMechanism().equals( XOAUTH ) ) {
```

Comment:
I had a comment earlier about using OOP. I wouldn't worry about that now for this pull request. 

We may have time later to refactor the code. I think it's important that we have the functionality in place to send Oauth Emails.

Normally when you write functions and you have if _condition-A_ do this implementation and if _condition-B_ do this other implementation in several places, it indicates you can create classes to keep the code modular.

Reply:
Please, see https://github.com/pentaho/pentaho-platform/pull/5185#discussion_r1086016229.